### ISSUE TITLE
Broken FastAPI dependency

### ISSUE THREAD

**Issue by jfeaver** [zmievsa/cadwyn#210]
**Describe the bug**
A recent minor version bump in FastAPi is a breaking change for Cadwyn. The Cadwn middleware calls `solve_dependencies` which now requires a named argument: `embed_body_fields`. (0.112.4 release of FastAPI): https://github.com/fastapi/fastapi/pull/12117/files#diff-aef3dac481b68359f4edd6974fa3a047cfde595254a4567a560cebc9ccb0673fR553)

Stacktrace:
```
ERROR:    Exception in ASGI application
  + Exception Group Traceback (most recent call last):
  |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_utils.py", line 77, in collapse_excgroups
  |     yield
  |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/base.py", line 186, in __call__
  |     async with anyio.create_task_group() as task_group:
  |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 680, in __aexit__
  |     raise BaseExceptionGroup(
  | ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
    |     result = await app(  # type: ignore[func-returns-value]
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    |     return await self.app(scope, receive, send)
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    |     raise exc
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    |     await self.app(scope, receive, _send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    |     await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    |     raise exc
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    |     await app(scope, receive, sender)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    |     await route.handle(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/routing.py", line 460, in handle
    |     await self.app(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    |     await super().__call__(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    |     await self.middleware_stack(scope, receive, send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    |     raise exc
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    |     await self.app(scope, receive, _send)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
    |     with collapse_excgroups():
    |   File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/contextlib.py", line 158, in __exit__
    |     self.gen.throw(value)
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_utils.py", line 83, in collapse_excgroups
    |     raise exc
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
    |     response = await self.dispatch_func(request, call_next)
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/cadwyn/middleware.py", line 63, in dispatch
    |     solved_result = await solve_dependencies(
    |                           ^^^^^^^^^^^^^^^^^^^
    | TypeError: solve_dependencies() missing 1 required keyword-only argument: 'embed_body_fields'
    +------------------------------------

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    raise exc
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/routing.py", line 460, in handle
    await self.app(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/base.py", line 185, in __call__
    with collapse_excgroups():
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/_utils.py", line 83, in collapse_excgroups
    raise exc
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/nathan/Library/Caches/pypoetry/virtualenvs/public-api-odHRiFV8-py3.12/lib/python3.12/site-packages/cadwyn/middleware.py", line 63, in dispatch
    solved_result = await solve_dependencies(
                          ^^^^^^^^^^^^^^^^^^^
TypeError: solve_dependencies() missing 1 required keyword-only argument: 'embed_body_fields'
```

**To Reproduce**
1. Create a Cadwyn app using FastAPI v0.112.4 or higher
2. Send a request to any endpoint (I noticed it testing a new POST request)
3. You should get a 500 error and see a similar stack trace in the server logs.

**Expected behavior**
The endpoint should return the usual response.

**Operating system**
MacOS 14.3.1 (23D60)

**Comment by zmievsa** [comment]
Thanks for your report! Will publish a fix within an hour.

**Comment by zmievsa** [comment]
Fixed in 4.2.3. Feel free to open other tickets, pull requests, or leave any feedback regarding Cadwyn in the future! 
https://pypi.org/project/cadwyn/4.2.3/
